### PR TITLE
SAMZA-2574 : improve flexibility of SystemFactory interface

### PR DESCRIPTION
**Pull request by MabelYC**
This is second pr to supplement commit: 4d97a8d93c47d629072648ac98dabd6a40746321

 
Changes: Added addition label info to SystemAdmin to help user provide more information to SystemFactory.
API Changes: Added several new functions to help provide more parameter. Won't impact current users.
Upgrade Instructions: None
Usage Instructions: None

### CODE REVIEW

File: samza-core/src/main/java/org/apache/samza/config/SystemConfig.java [hunk 1]
```diff
@@ -106,6 +123,19 @@ public SystemAdmin getSystemAdmin(String systemName) {
...
+   * Get {@link SystemAdmin} instance for given system name with an adminLabel.
+   * This function provides an extra input parameter to {@link #getSystemAdmin}, which can be used to provide extra
+   * information for the admin instance, e.g. ownership of client instance, to help better identify admins in logs,
+   * threads and client instances etc., along with other relevant information like systemName.
+   *
+   * @param systemName System name
+   * @return SystemAdmin of the system if it exists, otherwise null.
+   */
+  public SystemAdmin getSystemAdmin(String systemName, String adminLabel) {
+    return getSystemAdmins(adminLabel).get(systemName);
```
Comment (sborya) [inline samza-core/src/main/java/org/apache/samza/config/SystemConfig.java:136]: I understand this is an existing code, but it seems weird that we create ALL admins EVERY time just to get a single one.
Either we should create a single Admin, or we can cache the results of the previous getAdmins().

File: samza-core/src/main/java/org/apache/samza/config/SystemConfig.java [hunk 2]
```diff
@@ -96,6 +96,23 @@ public SystemConfig(Config config) {
...
+  /**
+   * Get {@link SystemAdmin} instances for all the systems defined in this config and set with an adminLabel.
+   * This function provides an extra input parameter to {@link #getSystemAdmins}, which can be used to provide extra
+   * information for the admin instance, e.g. ownership of client instance, to help better identify admins in logs,
+   * threads and client instances etc., along with other relevant information like systemName.
+   *
+   * @param adminLabel a string to provide info the admin instance.
+   * @return map of system name to {@link SystemAdmin}
+   */
+  public Map<String, SystemAdmin> getSystemAdmins(String adminLabel) {
```
Comment (sborya) [inline samza-core/src/main/java/org/apache/samza/config/SystemConfig.java]: can we change implementation of getSystemAdmins() to {
  return getSystemAdmins("").
}

Comment (MabelYC) [reply samza-core/src/main/java/org/apache/samza/config/SystemConfig.java]: This file is updated. The version you reviewed is the outdated one. Please see: https://github.com/apache/samza/pull/1420/files#diff-2063da999ce81031b6015c31991f60c4R100

File: samza-core/src/main/java/org/apache/samza/config/SystemConfig.java [hunk 3]
```diff
@@ -89,11 +98,14 @@ public SystemConfig(Config config) {
    * @return map of system name to {@link SystemAdmin}
    */
   public Map<String, SystemAdmin> getSystemAdmins() {
```
Comment (sborya) [inline samza-core/src/main/java/org/apache/samza/config/SystemConfig.java:100]: I wonder if it is better to pass the admin label here in getSystemAdmins()?

**Pull request closed without merging**